### PR TITLE
refactor(azure): get locations with self session

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

This pull request includes changes to the `azure_provider.py` file within the `prowler/providers/azure` directory. The main focus of these changes is to simplify the `get_locations` method by removing the need to pass credentials as an argument and instead using the session directly.

Key changes include:

* Simplification of the `__init__` method by removing the credentials argument from the `get_locations` method call.
* Modification of the `get_locations` method to no longer require credentials as an argument and directly use the session for authentication.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/azure/azure_provider.py [hunk 1]
```diff
@@ -942,33 +942,29 @@ async def get_azure_identity():
...
-                    "Content-Type": "application/json",
-                }
-                response = requests.get(url, headers=headers)
-                if response.status_code == 200:
-                    data = response.json()
-                    for location in data["value"]:
-                        locations[display_name].append(location["name"])
+        credentials = self.session
+        locations = {}
+        token = credentials.get_token("https://management.azure.com/.default").token
```
Comment (jfagoagas) [inline prowler/providers/azure/azure_provider.py]: Can you do that with the Azure SDK?

File: prowler/providers/azure/azure_provider.py [hunk 2]
```diff
@@ -942,33 +942,29 @@ async def get_azure_identity():
...
-    def get_locations(self, credentials) -> dict[str, list[str]]:
+    def get_locations(self) -> dict[str, list[str]]:
         """
         Retrieves the locations available for each subscription using the provided credentials.
 
-        Args:
-            credentials: The credentials object used to authenticate the request.
-
         Returns:
             A dictionary containing the locations available for each subscription. The dictionary
```
Comment (jfagoagas) [inline prowler/providers/azure/azure_provider.py:950]: Can you add an example of the output?

File: prowler/providers/azure/azure_provider.py [hunk 3]
```diff
@@ -942,33 +942,29 @@ async def get_azure_identity():
...
+            url = f"https://management.azure.com/subscriptions/{subscription_id}/locations?api-version=2022-12-01"
+            headers = {
+                "Authorization": f"Bearer {token}",
+                "Content-Type": "application/json",
+            }
+            response = requests.get(url, headers=headers)
+            if response.status_code == 200:
+                data = response.json()
+                for location in data["value"]:
+                    locations[display_name].append(location["name"])
```
Comment (jfagoagas) [inline prowler/providers/azure/azure_provider.py]: What is this function used for? Can you review the output format?

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/azure/azure_provider.py (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/azure/azure_provider.py:950]: Done

File: prowler/providers/azure/azure_provider.py (hunk 1 above)
Comment (pedrooot) [reply prowler/providers/azure/azure_provider.py]: Done

File: prowler/providers/azure/azure_provider.py (hunk 3 above)
Comment (pedrooot) [reply prowler/providers/azure/azure_provider.py]: Output format added on the docstrings. This method it's used to set `_locations` for AzureProvider but this it's never used. We should discuss what to do with this attributes.

**Review by jfagoagas — DISMISSED** [review]

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**